### DISCUSSION TITLE
how to save the number of counted objects?

### DISCUSSION THREAD

**Discussion by StefanoGarzoni** [roboflow/notebooks#119] [Q&A] [answered]
hello, i took the code to count the objects i have in figure via a linecounter from this video: https://www.youtube.com/watch?v=Mi9iHFd0_Bo

I found it very fascinating, but I was wondering how and if it was possible to save the number of objects counted in real time on an external file or similar.
I tried to look at the linecounter function but didn't find much on the internet.
I ask for your help in doing this.
As usual thanks very much to anyone who replies for taking the time and help!
You are unique, thanks again!

**Comment by SkalskiP** [comment]
Hi @Stefano0513 👋🏻! The simplest idea I have is to create a simple `txt` file and append a new line to that file after every frame.

1. Create a function to append a new line to the file:

```python
def append_line_to_file(file_path: str, line: str) -> None:
    with open(file_path, "a") as file:
        file.write(line + "\n")
```

2. You can [access](https://github.com/roboflow/supervision/blob/e16cca43497a6274ba3e15c3ad4913a88978f753/supervision/detection/line_counter.py#L27) counter values like this:

```python
line_counter.in_count

line_counter.out_count
```

3. Put everything together:

```python
import cv2

from ultralytics import YOLO
import supervision as sv
import numpy as np


LINE_START = sv.Point(320, 0)
LINE_END = sv.Point(320, 480)


def main():
    line_counter = sv.LineZone(start=LINE_START, end=LINE_END)
    line_annotator = sv.LineZoneAnnotator(thickness=2, text_thickness=1, text_scale=0.5)
    box_annotator = sv.BoxAnnotator(
        thickness=2,
        text_thickness=1,
        text_scale=0.5
    )

    model = YOLO("yolov8l.pt")
    for result in model.track(source=0, show=True, stream=True, agnostic_nms=True):
        
        frame = result.orig_img
        detections = sv.Detections.from_yolov8(result)

        if result.boxes.id is not None:
            detections.tracker_id = result.boxes.id.cpu().numpy().astype(int)
        
        detections = detections[(detections.class_id != 60) & (detections.class_id != 0)]

        labels = [
            f"{tracker_id} {model.model.names[class_id]} {confidence:0.2f}"
            for _, confidence, class_id, tracker_id
            in detections
        ]

        frame = box_annotator.annotate(
            scene=frame, 
            detections=detections,
            labels=labels
        )

        line_counter.trigger(detections=detections)
        line_annotator.annotate(frame=frame, line_counter=line_counter)
        append_line_to_file('counts.txt', f'{line_counter.in_count}, {line_counter.out_count}')

        cv2.imshow("yolov8", frame)

        if (cv2.waitKey(30) == 27):
            break


if __name__ == "__main__":
    main()
```

**Reply by StefanoGarzoni** [reply, marked as answer]
thank you very much!! May I ask where can I find a "documentation" about the linezone and all the other instances?

**Comment by SkalskiP** [comment]
We have [documentation](https://roboflow.github.io/supervision/), and you can learn here a lot about most of the stuff we support. Unfortunately, `LineZone` is not here... mostly before I plan to rewrite the API. Some design decisions were wrong.

**Reply by StefanoGarzoni** [reply]
what a shame. In any case, thanks again!
